feat: add Heurist embedding model

tsubasakong opened the pull request:
# Relates to:
<!-- Add relevant issue/ticket link -->

# Risks
Low - This PR adds a new embedding model provider (Heurist) with BAAI/bge-large-en-v1.5 as the default model. Changes are additive and follow existing patterns.

# Background

## What does this PR do?
This PR adds support for a new embedding provider called "Heurist" with the following changes:
1. Adds HEURIST_EMBEDDING_MODEL configuration with BAAI/bge-large-en-v1.5 as default
2. Integrates Heurist into the embedding provider system with 1024 dimension vectors
3. Updates model configurations to use meta-llama/llama-3.3-70b-instruct and FLUX.1-dev
4. Implements remote embedding functionality for the Heurist provider

## What kind of change is this?
Features (non-breaking change which adds functionality) - Adds new embedding provider capability

# Documentation changes needed?
My changes require a change to the project documentation to document the new Heurist embedding provider and its configuration options.

# Testing

## Where should a reviewer start?
1. Check the .env.example file for the new HEURIST_EMBEDDING_MODEL configuration
2. Review the embedding.ts changes for proper integration
3. Verify the model configuration updates in the Models section

## Detailed testing steps
1. Configure HEURIST_EMBEDDING_MODEL in environment
2. Test embedding generation with the new provider:
   - Verify 1024-dimension vectors are generated correctly
   - Confirm remote embedding functionality works
   - Check fallback to BGE when appropriate


## Screenshots
N/A - No UI changes involved

# Deploy Notes
The following environment variables need to be configured:
- USE_HEURIST_EMBEDDING = true
- HEURIST_EMBEDDING_MODEL=BAAI/bge-large-en-v1.5
- HEURIST_API_KEY (if using remote embeddings)

ag-wnl commented on packages/core/src/embedding.ts:
```diff
@@ -37,25 +38,30 @@ export const getEmbeddingConfig = (): EmbeddingConfig => ({
-                ? getEmbeddingModelSettings(ModelProviderName.GAIANET)
-                      .dimensions
-                : 384, // BGE
+                ? 768 // GaiaNet
```
@tsubasakong please ensure to continue using getEmbeddingModelSettings(ModelProviderName.GAIANET) for embedding dimension for gaianet

tsubasakong replied:
thanks! Updated in the latest commit! 👍

ag-wnl commented on packages/core/src/embedding.ts:
```diff
@@ -37,25 +38,30 @@ export const getEmbeddingConfig = (): EmbeddingConfig => ({
-                : 384, // BGE
+                ? 768 // GaiaNet
+                : settings.USE_HEURIST_EMBEDDING?.toLowerCase() === "true"
+                  ? 1024 // Heurist
```
To add embedding dimensions for a model (ex: heurist here), a better approach is to instead add the dimension by making a `ModelClass.EMBEDDING`, in     `ModelProviderName.HEURIST` and use the value using `getEmbeddingModelSettings`

tsubasakong replied:
thanks! Updated in the latest commit! 👍

ag-wnl commented on packages/core/src/embedding.ts:
```diff
@@ -37,25 +38,30 @@ export const getEmbeddingConfig = (): EmbeddingConfig => ({
               : settings.USE_GAIANET_EMBEDDING?.toLowerCase() === "true"
-                ? getEmbeddingModelSettings(ModelProviderName.GAIANET).name
-                : "BGE-small-en-v1.5",
+                ? settings.GAIANET_EMBEDDING_MODEL || "nomic-embed"
```
Continue using `getEmbeddingModelSettings(ModelProviderName.GAIANET).name` as it elegantly handles the fallback case

ag-wnl commented on packages/core/src/embedding.ts:
```diff
@@ -146,6 +153,8 @@ export function getEmbeddingZeroVector(): number[] {
             ModelProviderName.GAIANET
         ).dimensions; // GaiaNet dimension
+    } else if (settings.USE_HEURIST_EMBEDDING?.toLowerCase() === "true") {
+        embeddingDimension = 1024; // Heurist dimension
```
^ use value using `getEmbeddingModelSettings` after adding it to model configuration as mentioned above

tsubasakong replied:
thanks! Updated in the latest commit! 👍

0xnogo committed:
feat: Add new translations and update README structure

- Introduced new language translations for Eliza in Arabic, Chinese, German, Spanish, French, Hebrew, Hungarian, Italian, Japanese, Korean, Dutch, Polish, Portuguese, Romanian, Russian, Serbian, Tagalog, Thai, Turkish, and Vietnamese.
- Updated the README structure to include links to the new translations.
- Added a new plugin '@elizaos/plugin-hyperliquid' in the pnpm-lock.yaml file with its dependencies.
- Adjusted the README links to point to the correct paths for translations.

Lukapetro committed:
fix: env variable naming

KacperKoza343 committed:
fix: resolve Windows path issue in pnpm build client

Fixes #2223

YoungPhlo committed:
docs: Add "What Did You Get Done This Week? 9" notes

swizzmagik committed:
Merge branch 'develop' into feat/add-birdeye-plugin

MonteCrypto999 committed:
Add a service example

augchan42 committed:
git commit -m "feat(echochambers): add dead room detection and conversation starter

- Add multiple room support
- Implement proactive conversation starter for inactive rooms
- Add configurable quiet period and check intervals
- Add debug logging for conversation starter process

Closes #2246"

wtfsayo committed:
Merge pull request #2228 from Stumble/develop

feat: add support for gravity chain in EVM plugin

wtfsayo committed:
Merge branch 'develop' into docs/community-streams

shakkernerd committed:
Merge pull request #2243 from YoungPhlo/docs/community-streams

docs: Add "What Did You Get Done This Week? #9" notes

wtfsayo committed:
Merge branch 'develop' into agw_integration

Hack666r committed:
Minor typo in CHANGELOG.md

shakkernerd committed:
chore: remove multiple langdetect package

Commit:
Typo fix README.md

shakkernerd committed:
chore: console -> elizaLogger

cygaar committed:
ts ignore

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2207 from cygaar/agw_integration

feat: Add AGW support to the Abstract plugin

oxlupo committed:
Add Persian README file

oxlupo committed:
Add Persian README file

odilitime committed:
Merge pull request #2260 from oxlupo/farsi-readme

docs: Add Farsi (Persian )readme

0xCardiE committed:
add price per address functionality

0xCardiE committed:
Merge branch 'develop' into feature/gecko_price_per_addy

mcp97 committed:
Merge pull request #2262 from 0xCardinalError/feature/gecko_price_per_addy

feat: CoinGecko - add price per address functionality

augchan42 committed:
git commit -m "fix(db): add limit param to memory retrieval across adapters

- Add limit parameter to getMemoriesByRoomIds in SQLite adapter
- Add limit parameter to getMemoriesByRoomIds in SQLjs adapter
- Add limit parameter to getMemoriesByRoomIds in PGLite adapter
- Add limit parameter to getMemoriesByRoomIds in Supabase adapter
- Fix query parameter ordering in SQLjs adapter
- Add consistent DESC ordering across all adapters

Closes #2253"

Commit:
add POST /agents and DELETE /agents/:agentId

odilitime committed:
fix: test/lint develop (#2266)

* bump version

* add eslint config

* remove unused error var per lint

* fix lint

* comment out duplicate symbol

* prefix address as it's not used per lint

* add _ prefix for unused vars per lint

* remove unused error per lint

* prefix unused vars with _ per lint

* remove unused error per lint

* don't import unused ServiceType per lint

* remove unused vars per lint

* rename test per lint

* lint auto fixes

* deTS it per lint

* deTS per lint

* bump

Commit:
feat: Adding plugin for B² Network (#2010)

* feat: b2-network plugin

* docs:update readme and package description

* fix: solve build error

* modify

* modify

* feat: transfer action

* modify README.md

* docs: update readme

* chore: update env file

* modify b2 plugin

* modify b2 plugin

* modify b2 plugin

* modify b2 plugin

* modify b2 plugin

* modify b2 plugin

* fix: default empty value and resolve type error

---------

Co-authored-by: goalongway <goalongway@163.com>
Co-authored-by: goalongway <36787789+goalongway@users.noreply.github.com>
Co-authored-by: goalongway <ma_huaibo@163.com>

shakkernerd committed:
Merge branch 'develop' into bugfix/optimize-recent-interactions-query

shakkernerd committed:
Merge pull request #2264 from augchan42/bugfix/optimize-recent-interactions-query

fix: (db) add limit param to memory retrieval across adapters

asyncjason committed:
Merge branch 'develop' into direct-client_agent-crud

odilitime committed:
Merge pull request #2256 from VitalikBerashvili/fix/typo

Typo fix README.md

odilitime committed:
Merge pull request #2255 from Hack666r/fix/typo

Minor typo in CHANGELOG.md

shakkernerd committed:
fix: add missing @elizaos/plugin-b2 to package.json

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2268 from elizaOS/fix/plugin-b2

fix: missing @elizaos/plugin-b2

odilitime committed:
Merge pull request #2232 from HowieDuhzit/develop

Add character creation template function to start.sh

odilitime committed:
Merge branch 'develop' into feat/add-birdeye-plugin

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

odilitime committed:
Merge branch 'develop' into koswald/voyageai

odilitime committed:
Merge pull request #2240 from blockydevs/2223--fix-pnpm-build-client-path

fix: resolve Windows path issue in pnpm build client

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

odilitime committed:
Merge branch 'develop' into feature/squid-bridge

odilitime committed:
Merge branch 'develop' into koswald/voyageai

odilitime committed:
Merge pull request #1482 from Archethect/feature/squid-bridge

feat: Add cross chain swaps through Squid Router

odilitime committed:
Merge pull request #1710 from thetechnocratic/plugin-evm-oz-governance

feat: Plugin evm oz governance

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

odilitime committed:
Merge pull request #1442 from Firbydude/koswald/voyageai

feat: Add support for VoyageAI embeddings API

odilitime committed:
Merge pull request #1417 from swizzmagik/feat/add-birdeye-plugin

feat: add birdeye plugin

wtfsayo committed:
Merge branch 'develop' into refactor-farcaster-client

sin-bufan committed:
refactor: simplify memory structure in FarcasterInteractionManager

- Removed the hash property from the memory content object.
- Updated the way memory is passed to the runtime processActions method to include the cast property in the content object.

asyncjason committed:
Merge branch 'develop' into direct-client_agent-crud

Commit:
don't need create, set works for this. catch startAgent errors. add PREVENT_UNHANDLED_EXIT support

Commit:
Merge branch 'direct-client_agent-crud' of https://github.com/jseparovic/eliza into direct-client_agent-crud

Commit:
fix merge issues. add back comments

lincheoll committed:
Fix: IME causes multiple messages on Enter (Fixes #2272)

lincheoll committed:
Merge branch 'develop' into 2272--fix-ime-multiple-messages

chuasonglin1995 committed:
Merge remote-tracking branch 'upstream/develop' into refactor-websearch

v1xingyue committed:
support load character from  character_url

v1xingyue committed:
Merge branch 'develop' into character-from-url

YoungPhlo committed:
docs: Add weekly contributor meeting notes

Commit:
feat: add getMemoryByIds to database adapters

Lukapetro committed:
chore: resolve merge conflicts with develop

shakkernerd committed:
Revert "feat: Add support for VoyageAI embeddings API"

shakkernerd committed:
Merge pull request #2290 from elizaOS/revert-1442-koswald/voyageai

Revert "feat: Add support for VoyageAI embeddings API"

wtfsayo committed:
Merge branch 'develop' into docs/contributors-meetings

wtfsayo committed:
Merge pull request #2285 from YoungPhlo/docs/contributors-meetings

docs: Add weekly contributor meeting notes

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: export b2Plugin

shakkernerd committed:
Merge branch 'develop' into fixes

shakkernerd committed:
Merge pull request #2291 from elizaOS/fixes

Fixes

wtfsayo committed:
Merge branch 'develop' into direct-client_agent-crud

v1xingyue committed:
Merge branch 'develop' into character-from-url

wtfsayo committed:
Merge pull request #2249 from PIPELINE-BTC/feat-add-a-service-example

feat: Add an example service

0xCardiE committed:
add all code

0xCardiE committed:
Merge branch 'develop' into feat/gitcoin_passport_plugin

v1xingyue committed:
Merge branch 'develop' into character-from-url

wtfsayo committed:
Merge pull request #2296 from 0xCardinalError/feat/gitcoin_passport_plugin

feat: Gitcoin passport

tcm390 committed:
Merge pull request #2274 from lincheoll/2272--fix-ime-multiple-messages

Fix: IME causes multiple messages on Enter (Fixes #2272)

HashWarlock committed:
fix: fix derive key to use path instead of subject for salt, update ra quote for verifiability

HashWarlock committed:
fix unit tests and update package json

shakkernerd committed:
Merge pull request #2303 from Phala-Network/fix/ra-and-derive-key

fix: derive key and update remote attestation

wtfsayo committed:
Merge branch 'develop' into direct-client_agent-crud

wtfsayo committed:
Merge pull request #2267 from jseparovic/direct-client_agent-crud

Improvement: Direct Client API - Add Delete Agent functionality

francis2tm committed:
feat: add new model classes for embedding and image processing

francis2tm committed:
feat: extend handleProvider function to include runtime, context, and modelClass options

CryptoGraffe committed:
fix farcaster memory

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

shakkernerd committed:
chore: console -> elizaLogger

lalalune committed:
Update README.md

lalalune committed:
Merge pull request #2309 from elizaOS/lalalune-patch-2

Update README.md

shakkernerd committed:
Merge branch 'develop' into pr/CryptoGraffe/2307

wtfsayo committed:
Merge branch 'develop' into refactor-farcaster-client

wtfsayo committed:
Merge branch 'develop' into 1186-add-get-memories-by-ids

wtfsayo committed:
Merge pull request #2293 from bbopar/1186-add-get-memories-by-ids

feat: add getMemoryByIds to database adapters

wtfsayo committed:
Merge branch 'develop' into character-from-url

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update .env.example

wtfsayo committed:
Merge pull request #2281 from v1xingyue/character-from-url

feat: support load character from  character_url

shakkernerd committed:
Merge branch 'develop' into pr/augchan42/2248

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2248 from augchan42/feature/echochamber-conversation-starter

feat: (echochambers) add dead room detection and conversation starter

wtfsayo committed:
chore: Update README.md (#2280)

* Update README.md

---------

Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

0xnogo committed:
moving new locales

0xnogo committed:
Remove swipr from git tracking

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

Lukapetro committed:
fix: remove goku fallback image, env variable renamed

Lukapetro committed:
chore: resolve merge conflicts with develop

Commit:
feat: support for eternalai provider can get onchain system prompt

Commit:
Merge remote-tracking branch 'origin/develop' into eternalai-agent

# Conflicts:
#	.env.example

shakkernerd committed:
Merge pull request #2087 from sin-bufan/refactor-farcaster-client

refactor: farcaster client env configuration

Lukapetro committed:
chore: resolve merge conflicts with develop

Commit:
Merge branch 'elizaOS:develop' into eternalai-agent

Commit:
feat: support for eternalai provider can get onchain system prompt

Commit:
Merge remote-tracking branch 'origin/eternalai-agent' into eternalai-agent

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

shakkernerd committed:
Merge pull request #2319 from eternalai-org/eternalai-agent

feat: Onchain Agent Transformer - transform any Eliza agents into unstoppable Solidity smart contracts deployed on 10+ blockchains

francis2tm committed:
feat: add Atoma SDK configuration to .env.example for Bearer Auth token

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

wtfsayo committed:
Merge branch 'develop' into pr/1865

wtfsayo committed:
Delete pnpm-lock.yaml

wtfsayo committed:
Create pnpm-lock.yaml

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update tokenProvider.ts

wtfsayo committed:
Update tokenProvider.ts

wtfsayo committed:
Merge pull request #1865 from hellopleasures/develop

feat: add Dex Screener plugin with token price action, evaluators, an…

tcm390 committed:
Merge branch 'develop' into refactor-websearch

wtfsayo committed:
Merge pull request #2195 from chuasonglin1995/refactor-websearch

refactor: websearch into a service

everimbaq committed:
fix: correct OPENROUTER_API_KEY env in config

Lukapetro committed:
Merge branch 'develop' into feature/add-client-instagram

shakkernerd committed:
Merge pull request #2324 from xNomad-AI/main

fix: correct OPENROUTER_API_KEY env in config

leeran7 committed:
Feat: Enables multiple remote urls

leeran7 committed:
Chore: Remove unneeded spaces

leeran7 committed:
Merge branch 'develop' into multiple-remote-urls

Lukapetro committed:
Merge branch 'develop' into feature/add-client-instagram

francis2tm committed:
feat: integrate Atoma model support with configuration updates and SDK version change

- Added Atoma API URL and model configurations to .env.example.
- Updated pnpm-lock.yaml to reference the latest Atoma SDK version.
- Enhanced index.ts to include Atoma as a model provider for token retrieval.
- Implemented Atoma model initialization and response handling in generation.ts.
- Updated models.ts to utilize dynamic settings for Atoma model classes.

francis2tm committed:
refactor: remove unused model configurations from .env.example and models.ts

- Deleted EMBEDDING_ATOMA_MODEL and IMAGE_ATOMA_MODEL from .env.example.
- Removed corresponding model class settings from models.ts to streamline configuration.

sukrucildirr committed:
Update README_TR.md

derrekcoleman committed:
Fix formatting in Browser section of README.md

Hey all! I saw strange formatting in the README.md and figured it was an extra three backticks ```

Sure enough, deleting those made the section render as intended. I also reworded one of the sentences that had three "and"s to make it a little more legible.

JoeyKhd committed:
upd: added initial docker image fixes

JoeyKhd committed:
upd: we need all packages in Dockerfile workdir

rferrari committed:
feat: groq image vision provider

JoeyKhd committed:
upd: reworked docker ignore

JoeyKhd committed:
upd: rebuilt docs and bumped version numbers

francis2tm committed:
Merge remote-tracking branch 'upstream/main' into feat/atoma-provider

francis2tm committed:
chore: update lockfile

francis2tm committed:
chore: update lockfile

francis2tm committed:
chore: update lockfile

ai16z-demirix committed:
Merge branch 'develop' into feature/solana-plugin-tests

JoeyKhd committed:
upd: resolved Dockerfile.docs build issues

JoeyKhd committed:
upd: reverted org generation in api docs

JoeyKhd committed:
upd: bumped types packages for docusaurus

thomasWos committed:
Update README.md

francis2tm committed:
refactor: restructure Atoma model configuration for improved clarity and consistency

- Updated Atoma model settings in models.ts to encapsulate parameters (stop, maxInputTokens, maxOutputTokens, temperature) within each model class.
- Reformatted code for better readability and maintainability.
- Minor formatting adjustments in types.ts for consistency in enum declaration and interface method signatures.

shakkernerd committed:
Merge branch 'develop' into pr/JoeyKhd/2347

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2347 from JoeyKhd/fix-dockerimage

fix: docker images+compose and broken documentation

francis2tm committed:
chore: restore original formatting

francis2tm committed:
chore: restore original formatting

francis2tm committed:
chore: restore original formatting

francis2tm committed:
chore: restore original formatting

francis2tm committed:
chore: revert types.ts

francis2tm committed:
Merge remote-tracking branch 'upstream/develop' into feat/atoma-provider

francis2tm committed:
refactor: improve code formatting and readability in generation.ts

- Standardized import statements and removed unnecessary line breaks.
- Simplified function definitions and improved inline comments for clarity.
- Enhanced logging messages for better debugging and error tracking.
- Ensured consistent use of single quotes for strings throughout the file.

francis2tm committed:
refactor: remove Atoma model generation function from generation.ts

- Deleted the handleAtoma function, which was responsible for generating objects for Atoma models.
- This change simplifies the codebase by removing unused functionality, improving maintainability.

francis2tm committed:
chore: revert deps

francis2tm committed:
chore: remove atoma-sdk

mcp97 committed:
Merge branch 'develop' into patch-2

mcp97 committed:
Merge pull request #2346 from thomasWos/patch-2

Fix github client README.md

mcp97 committed:
Merge pull request #2342 from rferrari/feature/groq-image-vision-provider

feat: groq image vision provider

mcp97 committed:
Merge pull request #2220 from ai16z-demirix/fix/issue-2164

improvement: using strict types to avoid erorrs like issue 2164

augchan42 committed:
feat: (ragKnowledge) Enhance RAG knowledge handling

- Added support for double-byte characters (e.g., Chinese) to ensure proper processing.
- Implemented cleanup of deleted knowledge files to maintain data integrity.
- Enabled loading knowledge from directories, simplifying configuration and management.

These improvements address performance and usability issues, facilitating better support for large knowledge sets.

Closes #2323

augchan42 committed:
Merge branch 'develop' into feature/rag-knowledge-improvements-v2

tcm390 committed:
Merge pull request #2336 from derrekcoleman/patch-1

Fix formatting in Browser section of README.md

tcm390 committed:
Merge pull request #2334 from sukrucildirr/develop

Update README_TR.md

C0ldSmi1e committed:
fix: align base url in client if api runs on a different port

odilitime committed:
Merge branch 'develop' into feature/rag-knowledge-improvements-v2

odilitime committed:
Merge pull request #2351 from augchan42/feature/rag-knowledge-improvements-v2

feat: (ragKnowledge) Enhance RAG knowledge handling

odilitime committed:
add back farcaster

HashWarlock committed:
fix: esure RA Action reply does not hallucinate

omahs committed:
Update README_FR.md

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

odilitime committed:
Merge pull request #1964 from Lukapetro/feature/add-client-instagram

feat: add instagram client

odilitime committed:
Merge pull request #2356 from omahs/patch-2

docs: Update README_FR.md

odilitime committed:
Merge pull request #2355 from elizaOS/HashWarlock-fix-ra-action-reply

fix: esure RA Action reply does not hallucinate

odilitime committed:
Merge pull request #2353 from C0ldSmi1e/develop

fix: align base url in client if api runs on a different port

odilitime committed:
Merge branch 'develop' into ng/moving-readmes

proteanx committed:
feat: add safe_mode (& cfg_scale) for venice image generation (#2354)

* venice: add safe_mode & cfg_scale for images

* add missing types

for safeMode & cfgScale

* update lockfile

---------

Co-authored-by: Sayo <hi@sayo.wtf>

kesar committed:
Update package.json with proper dependencies (#2301)

these packages are being used in prod build

Co-authored-by: Sayo <hi@sayo.wtf>

odilitime committed:
Merge pull request #2149 from 0xnogo/ng/moving-readmes

chore: Reorganizing README translations into a dedicated i18n directory structure

odilitime committed:
bump version

odilitime committed:
bump lockfile

odilitime committed:
Merge pull request #2359 from elizaOS/odi-fix

chore: Prep 0.1.9-alpha.1

JoeyKhd committed:
Merge pull request #4 from elizaOS/develop

merge dev

odilitime committed:
fix lint issues

odilitime committed:
Merge pull request #2328 from leeran7/multiple-remote-urls

feat: Enable Multiple remote character urls

odilitime committed:
Merge pull request #2368 from elizaOS/odi-fix

chore: lint

defitricks committed:
fix: Fix typo in import statement for Solana AgentKit plugin Update i… (#2370)

* fix: Fix typo in import statement for Solana AgentKit plugin Update index.ts

* Update index.ts

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>

0xCardiE committed:
feat: dexscreener trending (#2325)

* add to main index

* cleanup

---------

Co-authored-by: Sayo <hi@sayo.wtf>

JoeyKhd committed:
upd: added the possibility to configure S3 endpoint, whether SSL is enabled and the forceStylePath

tcm390 committed:
check if remote url is defined first (#2384)

0xSero committed:
Startup Error: ENV misconfig (#2378)

* fix

* Update info.json

---------

Co-authored-by: Sero <69639595+Seroxdesign@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

dev-whoan committed:
fix: check env REMOTE_CHARACTER_URLS is defined (#2382)

Co-authored-by: Sayo <hi@sayo.wtf>

ericlehong committed:
Fix typo (#2385)

* fix: typo

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>

* fix: typo

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>

---------

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

Archethect committed:
fix: set default Squid Router throttle to 1000 (#2386)

crStiv committed:
chore: improve descriptions  (#2394)

* Update index.md

* Update quickstart.md

* Update agents.md

* Update actions.md

* chore: with -> through

shakkernerd committed:
Merge branch 'develop' into feature/solana-plugin-tests

shakkernerd committed:
Merge pull request #2345 from ai16z-demirix/feature/solana-plugin-tests

feat: adding tests for plugin-solana

Commit:
enhancement: invoke exit code 0 after positive cleanup

shakkernerd committed:
Merge branch 'develop' into pr/francis2tm/2082

shakkernerd committed:
Merge pull request #2082 from francis2tm/feat/atoma-provider

feat: atoma provider

shakkernerd committed:
Merge branch 'develop' into cleanup-exit-code

shakkernerd committed:
Merge pull request #2398 from erise133/cleanup-exit-code

chore: invoke exit code 0 after positive cleanup

shakkernerd committed:
feat: add eslint config file

shakkernerd committed:
fix: let -> const

shakkernerd committed:
fix: remove unused error var

shakkernerd committed:
fix: unused declaration of var

shakkernerd committed:
Merge pull request #2400 from elizaOS/fixes

fix: lint errors

ai16z-demirix committed:
feat: adding tests for slack client. Moving existing tests to new __tests__ directory.

ai16z-demirix committed:
feat: adding tests for github client

shakkernerd committed:
Merge pull request #2404 from ai16z-demirix/test/client-slack

feat: adding tests for slack client. Moving existing tests to new __tests__ directory.

shakkernerd committed:
Merge branch 'develop' into test/client-github

odilitime committed:
Merge pull request #2379 from JoeyKhd/s3-improvement

feat: better S3 flexibility

shakkernerd committed:
Merge pull request #2407 from ai16z-demirix/test/client-github

feat: adding tests for github client

enigmarikki committed:
feat: injective plugin (#1764)

* "plugin: init injective integration and skeleton"

* "wip : skeleton partially done"

* "wip: finished fetch"

* "chore : fix format"

* "chore : renamed files"

* "wip: completed grpc endpoints for onchain queries"

* "chore : removed unwanted files"

* "wip : adding templates + integrating onchain functions"

* "wip : added inital version of templates"

* "wip : testing the plugin"

* "wip : skeleton for actions, providers"

* "chore: renamed and fixed relative imports"

* "wip : added message broadcasting rpcs"

* "feat : finished initial version of message parsing and querying"

* "chore : refactor package.json"

* "chore: added chain message broadcasters to the InjectiveGrpcBase"

* "wip : restructure actions and providers"

* "chore : fix conflicts on lockfile"

* "chore: added readme.md to use injective-sdk-client-ts"

* "chore : renamed the package to ensure consistency and updated the docs"

* "chore : refactored function args to make it easy to define function schema"

* "fix : missing imports for wasm"

* "wip : refactor injective types and templates"

* "chore : refactored the types into different files for maintainability"

* "chore : fmt"

* "wip : refactor the responses for standard message parsing"

* "chore : refactored all modules to ouput same a common message type"

* "wip : minimal tests"

* "fix: the grpc init + bug in making request"

* "wip : done with skeleton templates"

* "chore : reorgranize files"

* "wip : adding exchange templates"

* "chore : added fmt and finished comprehensive template for exchange module"

* "wip: refactored templates for auction and bank"

* "wip : almost done refactoring all the templates to add both the request, response"

* "chore : renamed to gov for consistency and added the explorer module"

* "fix : removed minor import into exchange"

* "wip : added the base action"

* "feat: finished integrating actions, wip fix for templates and examples"

* "chore : refactored all the request and response templates"

* "chore : fmt templates"

* "chore : added modules examples except exchange, fixed a few relative imports"

* "chore: finished integrating exchange examples"

* "feat: finished all integrations for injective-plugin"

* "fix : revert fmt"

* "chore : bump injective-ts sdk version"

* "fix : dependency issues"

* "fix : complex dependency issue from esm imports"

* "chore : refactored the rpc to take either inj address or eth address"

* "wip: fixed rpc initialization"

* "fix: Refactored action template"

* "chore : resolve lock file conflicts"

* "chore : added config params in .env.examples"

* "wip : improve response prompts from agent"

* "chore : refactored similes, for better interpretation"

* "wip: integrated mito's constant product market maker - amm"

* "feat: added Initial Dex offering and action, for agents to autonomously raise funds"

* "chore : revert defaultCharacter to LLAMA"

---------

Co-authored-by: rikki <rikki@mahishasuramardini.mynetworksettings.com>
Co-authored-by: Odilitime <janesmith@airmail.cc>

its-DeFine committed:
feat: Integrate Livepeer LLM provider  (#2154)

* add livepeer on index.ts as llm provider

* updated livepeer models

* add livepeer as llm provider

* add retry logic on livepeer img gen

* add handlelivepeer

* update test

* add livepeer model keys on .example.env

* Merge pull request #2 from Titan-Node/livepeer-doc-updates

Updated docs for Livepeer LLM integration

* add endpoint on livepeer on models.ts

* edit livepeer model config at model.ts

* Add Livepeer to image gen plugin environments

Fixes this error
```
Error handling message: Error: Image generation configuration validation failed:
: At least one of ANTHROPIC_API_KEY, NINETEEN_AI_API_KEY, TOGETHER_API_KEY, HEURIST_API_KEY, FAL_API_KEY, OPENAI_API_KEY or VENICE_API_KEY is required
    at validateImageGenConfig (file:///root/eliza-test/eliza-livepeer-integration/packages/plugin-image-generation/dist/index.js:38:19)
```

* add comments on livepeer model sizes

* remove retry logic from livepeer generate text and img

* Fixed .env naming convention and fixed mismatch bug within code

* add bearer on livepeer calls

* change in parsing to accomodate for new livepeer update

* addadd nineteen api key on the message

---------

Co-authored-by: Titan Node <hello@titan-node.com>

wtfsayo committed:
Delete tate.character.json (#2425)

twilwa committed:
refactor: dockerize smoke tests (#2420)

* add workflow to block minified JS

* modify smoketests to run in a container, add caching

* use bullseye for bash

* node 23

---------

Co-authored-by: Sayo <hi@sayo.wtf>

sweetmantech committed:
feat: coinbase agentkit plugin for eliza (#2298)

* I add files for plugin-agentkit

* instructions added to gitignore.

* I update package name.

* I add updated files.

* I add ENV requirements to prevent default Eliza bugs.

* rollback index

* simple agentkit import.

* I load existing wallet data for provider.

* remove dev logging

* Update pnpm-lock.yaml

* update .env.example and support non 'base-sepolia'

* Update pnpm-lock.yaml

* Update package lock for smoke tests.

* I rollback changes to pnpm-lock file.

* fixing smoke tests for pnpm missing

* rollback changes to agent start file.

* I rollback changes to smoke test.

* Update pnpm-lock.yaml

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

tsubasakong committed:
feat:update heurist env var examples

tsubasakong committed:
Merge branch 'develop' into develop

twilwa committed:
feat: add workflow to block minified JS (#2417)

* add workflow to block minified JS

* Update block-mini.yml

print violating filenames and line numbers

* Modify threshold

zo-sol committed:
feat: Code In Plugin, load characters from blockchain (#2371)

* add ia

* Delete pnpm-lock.yaml

* Update agent/src/index.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* Update packages/plugin-iq6900/package.json

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* Update agent/src/index.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* update solana version as others

* Update package.json

* test iq6900 env issue

* fix env file and lock

* edited require

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

YoungPhlo committed:
docs: Add notes for weekly contributor meeting on 2025-01-14 (#2426)

Co-authored-by: Sayo <hi@sayo.wtf>

tcm390 committed:
fix: Fix Incorrect Tweet ID Parameter Passed to sendTweet Function (#2430)

* correct id

* skip action if model is not supported

* warn

* update warn

wtfsayo committed:
Update quickstart.md

tcm390 committed:
Merge pull request #2437 from elizaOS/chore-remove-merge-conflict-in-quickstart

fix: quickstart.md conflicts

Noisyxl committed:
Update README.md (#2438)

leopardracer committed:
Update seed.sql (#2435)

Co-authored-by: Sayo <hi@sayo.wtf>

AIFlowML committed:
feat: Pyth Data Plugin  (#2434)

* feat(pyth): add Pyth Data plugin

- Create new plugin for Pyth Data integration
- Add EventSource client for price feeds
- Configure plugin in agent
- Update dependencies and configuration files

This plugin enables real-time price feed data from Pyth Network,
including price updates, TWAPs, and publisher caps information.

* feat: Wrap up changes for Pyth Data integration

* Save local changes before merging upstream

* Update index.ts, package.json, and add .env.example for Pyth Data integration

* Missing lock

---------

Co-authored-by: Sayo <hi@sayo.wtf>

JoeyKhd committed:
upd: if VITE_SERVER_PORT is not defined, use default 3000 (#2433)

Co-authored-by: Sayo <hi@sayo.wtf>

Hopium21 committed:
Update README.md (#2441)

Co-authored-by: Sayo <hi@sayo.wtf>

KacperKoza343 committed:
feat: ibc transfer on cosmos blockchains (#2358)

* feat: ibc transfer action

* refactor: remove redundant services and use skipClient

* update: eliza package import

* feat: add bridge data fetcher and update tests

* update: add IBC transfer action to README

* refactor: make bridge data fetcher reusable across the project

* fix: update similes for ibc transfer action

* fix: update toAddress in ibc transfer

* refactor: add changes according to review

* fix: ibc transfer action

bundinho committed:
fix TEE Log plugin errors at agent startup (#2415)

* fix TEE Log configuration error

* fix error telling 'Cannot open database because the directory does not exist'

* fix error telling 'Invalid TEE_MODE: undefined. Must be one of: LOCAL, DOCKER, PRODUCTION' at start up

wtfsayo committed:
Update index.ts (#2444)

JoeyKhd committed:
feat: improvement to logger (#2396)

* upd: initial updated logger

* upd: remove test case

* upd: improve some logs

* upd: resolved issues with mix of left and right json

* fix env and default mode

* Update pnpm-lock.yaml

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>

wtfsayo committed:
Merge branch 'develop' into develop
